base/process.jl

```diff
@@ -335,6 +335,20 @@ end
 pipe_reader(p::Process) = p.out
 pipe_writer(p::Process) = p.in
 
+"""
+    getpid(p::Process) -> Int32
+
+Get the process ID of a running process. Returns `-1` if the
+process is not running.
+"""
```

Comment:
What about alternatives when the process is not running, like throwing, or returning `nothing` ?

Reply:
Returning `nothing` would violate type stability.  Throwing is an option, but feel returning `-1`is more convenient.  There was also a suggestion to save the `pid` so that it can be inspected after the process has ended.  I don't think this is a good idea because pids can be reused and having `getpid(p)` return a potentially unrelated pid is error prone.  (E.g. you don't want to accidentally kill an unrelated process.)

Reply:
I ran an experiment with node, and it looks like they cache the pid:

```javascript
> const { spawn } = require('child_process');
> x = spawn('sleep',  ['1']);
> x.exitCode
0
> x._handle
null
> x.pid
45660
```
I still don't think this is a good idea for the reasons I mentioned above.

Reply:
I think this is essentially the same question as https://github.com/JuliaLang/julia/issues/18145. But upon reflection, I think an error would be better here. In some places, `-1` is used as a special value to mean "all processes". So `kill(getpid(p))` could accidentally cause the system to reboot (surprise!) with this choice of sentinel value.

Reply:
Fair enough. I'll make the change.  Just to make sure: caching the pid is off the table.

Reply:
`nothing` may make it type unstable, but this seems to be the new way of returning what was a `Nullable` (ie. returning `Union{T, Void}` or `Union{Some{T},Void}`), and `getpid` is a function for which I would tend to return a `Nullable`.